### PR TITLE
Room conversation avatar management

### LINKED ISSUE #1075: DM chat showing wrong avatar. It's showing autogenerated one rather than actual avatar even though it loaded.
Image

Image

### PR DESCRIPTION
**Pull request by kumarpalsinh25**
This PR revised the Room conversation avatar management based on the different criteria.

- DM avatar issue is resolved. fixes #1075 #846 
- Group DM now have separate avatar UI

### CODE REVIEW

**Review by bitfriend — APPROVED** [review]
I confirmed that opponent avatar is shown for DM.
Great work.

File: .changes/1078-chat-avatar.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- [Fix] DM avatar was not showing properly
+- [Enhance] Now DM and GroupDM have seperate avatar views
```
Comment (gnunicorn) [inline .changes/1078-chat-avatar.md:2]: I know I am being pedantic here, but a bit more user-value-oriented please ;) .

```suggestion
- DM's now show the avatar of the user you are talking to, if no avatar was set.
- In Group DMs the Avatar is showing several users you are in the DM with.
```

File: app/lib/features/chat/widgets/room_avatar.dart [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+class RoomAvatar extends ConsumerWidget {
+  final String roomId;
+  final double avatarSize;
+
+  const RoomAvatar({
+    required this.roomId,
+    this.avatarSize = 36,
+    Key? key,
+  }) : super(key: key);
```
Comment (gnunicorn) [inline app/lib/features/chat/widgets/room_avatar.dart:16]: Isn't this sufficient here:
```suggestion
    Key? super.key,
  });
```

File: app/lib/features/chat/widgets/room_avatar.dart [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      data: (members) {
+        int count = members.length;
+
+        //Show member avatar
+        if (count == 1) {
+          return memberAvatar(members[0], ref);
+        }
+
+        //Show opponent member avatar
+        else if (count == 2) {
```
Comment (gnunicorn) [inline app/lib/features/chat/widgets/room_avatar.dart:83]: let's keep if and else if together please.
```suggestion
        } else if (count == 2) {
          //Show opponent member avatar
```

**Review by gnunicorn — APPROVED** [review]
Looks good to me, just some minor wording and style things.

**Pull request merged**